File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        mmm = max(piles)
        a = 1
        b = mmm
        res = []
        while a<=b:
            cur = 0
            mid = (a + b) // 2
            for i in range(len(piles)):
                rr = piles[i]//mid + (piles[i]%mid != 0) * 1
                if rr != 0:
                    cur += rr
                else:
                    cur += 1
            print(cur, mid)
            if cur <= h:
                b = mid - 1
                res.append(mid)
            if cur > h:
                a = mid + 1
        print(res)
        return min(res)

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_r = 0;
        int l = 0, r = 1;
        while(r < prices.size()){
            if (prices[l] < prices[r]){
                int cur  = prices[r] - prices[l];
                max_r = max(max_r, cur);
            } else {
                l = r;
            }
            r+=1;
        }
        return max_r;
    }
};
